File: 58. Length of Last Word/Solution.cs
public class Solution
{
    public int LengthOfLastWord(string s)
    {
        if (s.Length == 1) return s[0] == ' ' ? 0 : 1;
        int lastIndex = 0;
        bool flag = false;
        for (int i = s.Length - 1; i >= 0; i--)
        {
            if (flag && s[i] == ' ')
            {
                return lastIndex - i;
            }
            else if (!flag && s[i] != ' ')
            {
                flag = true;
                lastIndex = i;
            }
        }
        return flag ? lastIndex + 1 : 0;
    }
}

File: 58. Length of Last Word/Solution.java
class Solution {
    public int lengthOfLastWord(String s) {
        if (s.length() == 1)
            return s.charAt(0) == ' ' ? 0 : 1;

        int lastIndex = 0;
        boolean lastSpaceFound = false;

        for (int i = s.length() - 1; i >= 0; i--) {
            
            if (lastSpaceFound && s.charAt(i) == ' ')
                return lastIndex - i;
            else if (!lastSpaceFound && s.charAt(i) != ' ') {
                lastSpaceFound = true;
                lastIndex = i;
            }
        }
        return lastSpaceFound ? lastIndex + 1 : 0;
    }
}